.github/workflows/pr.yml

```diff
@@ -177,12 +183,64 @@ jobs:
       - name: Untar artifacts
         shell: bash
         run:  tar xfz target.tgz
-      - name: setup node
-        shell: bash
-        run: (cd ui/client; npm ci)
       - name: Prepare docker
         shell: bash
         run: echo "version in ThisBuild := \"$GITHUB_SHA\"" > version.sbt && ./ciRunSbt.sh dist/docker:publishLocal
       - name: Test docker
         shell: bash
-        run: cd ./demo/docker && echo "NUSSKNACKER_VERSION=$GITHUB_SHA" > .env && ./testQuickstart.sh
+        run: cd ./demo/docker && echo "NUSSKNACKER_VERSION=ghactionstest-$GITHUB_SHA" > .env && ./testQuickstart.sh
+
+#TODO: extract to different workflow?
+  publish:
+    runs-on: ubuntu-latest
+    needs: [build, tests, crossCompile, integrationTests, slowTests, frontendTests, dockerTest]
+    if: ${{ github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/master' || startsWith(github.head_ref, 'preview') ||  github.head_ref == 'publish_githubactions' }}
+    steps:
+      - name: debug
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+
+      - uses: actions/checkout@v1
+      - name: Cache npm
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.11.0
+      - name: Setup npm
+        run: (cd ui/client; npm ci)
+      - name: Cache ivy packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/*.sbt') }}
+          restore-keys: ${{ runner.os }}-sbt
+      - uses: olafurpg/setup-scala@v10
+      - uses: actions/download-artifact@v2
+        with:
+          name: build-target
+      - name: Untar artifacts
```

Comment:
Do we really need it?

Reply:
well, caching makes build a bit faster, tar is needed because upload files one by one is really slow as there is awful lot of them